EIPS/eip-7923.md

```diff
@@ -145,21 +147,19 @@ index bf34fbee..477f969e 100644
 +        start_page = start_position >> 12
 +        end_page = end >> 12
 +
-+        gas = 0
-+
 +        for page in range(start_page, end_page + 1):
 +            if page not in self._memory.pages:
-+                gas += ALLOCATE_PAGE_COST
++                reason = f"Allocating page {page << 12}"
++                self._gas_meter.consume_gas(ALLOCATE_PAGE_COST, reason)
```

Comment:
Hmm, just realized, but the page is never added to `self._memory.pages` so this will keep allocating the page even if it's already there (was also the case before this commit)

Reply:
right, nice catch

Reply:
eb887705ac365ef617cfb7e28abe3815c7adadaa